llvm/lib/IR/AutoUpgrade.cpp

```diff
@@ -1349,6 +1349,10 @@ static bool upgradeIntrinsicFunction1(Function *F, Function *&NewFn,
       else if (Name == "clz.ll" || Name == "popc.ll" || Name == "h2f" ||
                Name == "swap.lo.hi.b64")
         Expand = true;
+      else if (Name == "barrier0" || Name == "barrier.n" ||
```

Comment:
I look at this `else if()` sequence with fixed string comparisons and can't help but wonder if most of them can be consolidated into a single `StringSwitch`.

Reply:
I'm moved these cases as well as the others without a consume_front into a StringSwitch in the else branch. I agree we could go further but looks like we used to have what you described and switched to this style in https://github.com/llvm/llvm-project/commit/b045c36ab92f4ff478dab678aaff4680fbccc7ea so maybe there are arguments for both?

Reply:
Ideally, all we need is a "has_known_prefix" function and a sorted array of known prefixes. This would give us both efficient search and an easily readable/maintainable list of intrinsics we care about. But given that it's nowhere near the hot path, we don't have to do it. The current version is fine.